### ISSUE TITLE
CrossEntropyLoss missing shape/type validation for 1D float targets

### ISSUE THREAD

**Issue by UjasShah** [pytorch/pytorch#174943]
### 🐛 Describe the bug

When running on MPS, nn.CrossEntropyLoss() does not raise an error when 1D labels are passed in as floats. Even accepts labels like 1.2 (shown below). When run on CPU, it does produce an error as expected.

```
import torch
import torch.nn as nn

def reproduce_issue(device_name):
    device = torch.device(device_name)
    print(f"\n--- Testing on {device_name.upper()} ---")
    
    criterion = nn.CrossEntropyLoss() # 2 classes, batch size of 3
    logits = torch.randn(3, 2, requires_grad=True).to(device)
    
    # CASE: 1D Float Labels (Invalid for standard indexing, should be Long)
    labels = torch.tensor([0.0, 1.2, 0.0], dtype=torch.float).to(device)
    
    print(f"Logits shape: {logits.shape}, dtype: {logits.dtype}")
    print(f"Labels shape: {labels.shape}, dtype: {labels.dtype}")
    
    # Raises an error on CPU but passes silently on MPS
    try:
        loss = criterion(logits, labels)
        print(f"SUCCESS: Loss calculated as {loss.item():.4f}")
        loss.backward()
        print("SUCCESS: Backward pass completed.")
    except Exception as e:
        print(f"CAUGHT EXPECTED ERROR: {e}")

# Test CPU behavior (Expected: Raise Error)
reproduce_issue("cpu")

# Test MPS behavior (Expected: Should also Raise Error, but currently passes)
if torch.backends.mps.is_available():
    reproduce_issue("mps")
else:
    print("\nMPS not available on this machine.")


Below is the output
--- Testing on CPU ---
Logits shape: torch.Size([3, 2]), dtype: torch.float32
Labels shape: torch.Size([3]), dtype: torch.float32
CAUGHT EXPECTED ERROR: expected scalar type Long but found Float

--- Testing on MPS ---
Logits shape: torch.Size([3, 2]), dtype: torch.float32
Labels shape: torch.Size([3]), dtype: torch.float32
SUCCESS: Loss calculated as 1.0991
SUCCESS: Backward pass completed.
```

### Versions

PyTorch version: 2.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 26.2 (arm64)
GCC version: Could not collect
Clang version: 17.0.0 (clang-1700.6.3.2)
CMake version: Could not collect
Libc version: N/A

Python version: 3.13.3 (main, Apr  8 2025, 13:54:08) [Clang 16.0.0 (clang-1600.0.26.6)] (64-bit runtime)
Python platform: macOS-26.2-arm64-arm-64bit-Mach-O
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Apple M3

Versions of relevant libraries:
[pip3] numpy==2.4.2
[pip3] torch==2.10.0
[conda] Could not collect

cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk
